Fix shairport

hgy59 opened the pull request:
_Motivation:_  Shairport-sync DSM6 compatibiliy is asked a long time ago
_Linked issues:_  #3040 (with links to other issues), fixes #3922

### Checklist
- [x] Build rule `all-supported` completed successfully (except for ppc853x-5.2)
- [ ] Package upgrade completed successfully (hard to test - old package does not install, may be test with upgrade 2.3.11 => 2.8.5)
- [x] New installation of package completed successfully (except DSM <6)
- [ ] Fix installation on DSM<6
- [x] Update shairport to version 2.8.5 (#2529) or later


### Remarks
- update busybox to enable --large-opts (yes I like readable parameters)
- busybox removed from dependencies and use daemonize of shairport

hgy59 committed:
fix shairport-sync service installation

- update to DSM6 service configuration
- fix pid file creation (configure option --with-piddir has no effect, default location is var/run)
- create start-stop-status functions base on projects systemv template
- add busybox module usrmng (otherwise busybox install fails for unknown reason)
- create soft links with busybox install
- include all documented settings in shaiport-sync.conf
- TODO: fix DSM5 installation

hgy59 committed:
update shairport-sync to version 2.8.6

hgy59 committed:
fix url for cross/popt

ymartin59 commented:
@hgy59 From my point of view, we may discard DSM 5 support here. Probably it helps to deliver this update.

hgy59 committed:
update shairport-sync to version 3.3.6

- use daemonize and remove busybox start-stop-service
- use generic SERVICE_COMMAND
- add patch to compile for ppc853x

ymartin59 commented on spk/shairport-sync/Makefile:
```diff
@@ -1,31 +1,28 @@
-INSTALL_PREFIX = /usr/local/$(SPK_NAME)
+# the path of the pid file must be configured in the configfile
+# (sessioncontrol.daemon_pid_dir) as "/usr/local/$(SPK_NAME)/var"
+SERVICE_COMMAND = $${SYNOPKG_PKGDEST}/bin/$(SPK_NAME) --daemon --configfile=/usr/local/$(SPK_NAME)/var/$(SPK_NAME).conf
```
Generally I prefer to refer to package with `/var/packages/$(SPK_NAME)/target/`

hgy59 committed:
incorporate review

cortices commented:
Nice job getting shairport-sync building happily :) looking forward to it landing in the package repo.

hgy59 commented:
> 
> Nice job getting shairport-sync building happily :) looking forward to it landing in the package repo.

@cortices can you test this package with your diskstation as I have no speaker to validate it is working?
And I can't see shairport in the network. Maybe some config is missing (either in shairport-sync or in the client).

I uploaded a pre-release in my forked repo [here](https://github.com/hgy59/spksrc/releases/tag/shairport-synch)

cortices commented:
I have installed the prerelease on my DS918+ (x64). It installs and runs fine and devices connect to it however there is no sound coming from the device. Confirmed using AirPlay from 1 Mac, 1 iPhone, and from the synology Audio Station itself. Not certain where I can locate clearer logs/enable verbose options, since `/var/packages/shairport-sync/target/var/shairport-sync.log` only logs startup and shutdown of the daemon, not connections. I'm also not sure where ALSA logs live.

hgy59 commented:
> 
> Not certain where I can locate clearer logs/enable verbose options, since `/var/packages/shairport-sync/target/var/shairport-sync.log` only logs startup and shutdown of the daemon, not connections. I'm also not sure where ALSA logs live.

Please have a look at `/var/packages/shairport-sync/target/var/shairport-sync.conf`.
There are several options you can set. Most of them are commented to show the default values.
In the diagnostics section you can set debug level (verbosity).


The alsa config files are located in `/var/packages/shairport-sync/target/share/alsa/`

Unfortunately there is no older shairport-sync package available that could be used to validate that it once worked as expected.

Validate ports with:
```
# netstat -tulpn | grep shairport
tcp        0      0 0.0.0.0:8303            0.0.0.0:*               LISTEN      28388/shairport-syn
tcp6       0      0 :::8303                 :::*                    LISTEN      28388/shairport-syn
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           28388/shairport-syn

```

As analyzing this, I found that port 5353 is not configured for DSM by now.
And there is an issue that is solved by opening port 5353 https://github.com/mikebrady/shairport-sync/issues/255

hgy59 commented:
I am updating the prerelease with the port 5353 fix [here](https://github.com/hgy59/spksrc/releases/tag/shairport-synch)
I deleted the former packages. x64 and 88f628 are already there.

hgy59 commented:
Sorry, the port 5353 fix was a bad solution. 
Meanwhile I got shairpoint-sync 2.3.11-2 (beta) installed from the synocommunity repo on my DS-210+ (ppc853x-5.2).
Still no audio output available, but audio station shows the installed shairport-sync as output device.

With this it sould be possible to find the delta to the updated version that is not found as output device yet.

cortices commented:
I've set the log verbosity to 2 and then 3 in the config file and restarted with `synopkg` but I'm still not seeing any real logging in the `target/var/shairport-sync.log` file. I have confirmed other config changes have had effect.

hgy59 commented:
To see some logging when I stop shairport-sync and manually run it on command line like this:
```bash
# cd /var/packages/shairport-sync/target/bin
# ./shairport-sync -v -v -v --configfile=../var/shairport-sync.conf
```

Then I can see errors, but have no idea what it is about:
```
         0.020528668 "tinysvcmdns.c:1504" data from=192.168.0.209 size=83
         0.000037892 "tinysvcmdns.c:756" rr_data_len goes beyond packet buffer: 25971 > 61
         0.000008138 "tinysvcmdns.c:894" error parsing answer #0
```

cortices commented:
Okay cool, I got proper logging by running it separately. When the AirPlay device connects there's a bunch of logs of them negotiating and setting parameters, and then the relevant errors begin:

````
ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
ALSA lib conf.c:4371:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4371:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
ALSA lib conf.c:4371:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:4850:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2450:(snd_pcm_open_noupdate) Unknown PCM default
         0.001903479 "audio_alsa.c:438" *warning: the alsa output_device "default" can not be found.
````

This block repeats as shairport-sync attempts to pipe control & PCM data to ALSA. Clearly I need to do some more research on how ALSA works on Synology's cutdown Debian.

hgy59 commented:
And this is the full log:
```
ash-4.3# ./shairport-sync -v -v -v   --configfile=../var/shairport-sync.conf  cd "`printf '%b' '\0057var\0057log'`"
         0.000225280 "shairport.c:435" Looking for configuration file at full path "/volume1/@appstore/shairport-sync/var/shairport-sync.conf"
         0.000967361 "shairport.c:1636" Started!
         0.000073010 "shairport.c:1660" software version: "3.3.6-libdaemon-OpenSSL-tinysvcmdns-ALSA-sysconfdir:/var/packages/shairport-sync/target/etc"
         0.000038404 "shairport.c:1666" log verbosity is 3.
         0.000054138 "audio_alsa.c:1001" alsa: init() -- alsa_backend_state => abm_disconnected.
         0.000045748 "audio_alsa.c:1018" alsa: alsa_maximum_stall_time of 0.200000 sec.
         0.000106952 "audio_alsa.c:1303" alsa: disable_standby_mode is "never".
         0.000045071 "audio_alsa.c:1305" alsa: disable_standby_mode_silence_threshold is 0.040000 seconds.
         0.000037637 "audio_alsa.c:1307" alsa: disable_standby_mode_silence_scan_interval is 0.004000 seconds.
         0.000058653 "audio_alsa.c:1346" alsa: output device name is "default".
         0.000197886 "shairport.c:1681" The processor is running little-endian.
         0.000040955 "shairport.c:1714" disable resend requests is off.
         0.000046865 "shairport.c:1717" diagnostic_drop_packet_fraction is 0.000000. A value of 0.0 means no packets will be dropped deliberately.
         0.000041087 "shairport.c:1718" statistics_requester status is 0.
         0.000048897 "shairport.c:1720" daemon status is 0.
         0.000057972 "shairport.c:1721" daemon pid file path is "/var/packages/shairport-sync/target/var/shairport-sync.pid".
         0.000053086 "shairport.c:1723" rtsp listening port is 8303.
         0.000050334 "shairport.c:1724" udp base port is 8304.
         0.000049897 "shairport.c:1725" udp port range is 10.
         0.000061421 "shairport.c:1726" player name is "*****".
         0.000039086 "shairport.c:1727" backend is "(null)".
         0.000051773 "shairport.c:1728" run_this_before_play_begins action is "(null)".
         0.000050673 "shairport.c:1729" run_this_after_play_ends action is "(null)".
         0.000038238 "shairport.c:1730" wait-cmd status is 0.
         0.000051172 "shairport.c:1731" run_this_before_play_begins may return output is 0.
         0.000050086 "shairport.c:1732" run_this_if_an_unfixable_error_is_detected action is "(null)".
         0.000050956 "shairport.c:1733" run_this_before_entering_active_state action is  "(null)".
         0.000060307 "shairport.c:1734" run_this_after_exiting_active_state action is  "(null)".
         0.000039132 "shairport.c:1735" active_state_timeout is  10.000000 seconds.
         0.000086978 "shairport.c:1736" mdns backend "(null)".
         0.000055251 "shairport.c:1737" userSuppliedLatency is 0.
         0.000050484 "shairport.c:1740" interpolation setting is "basic".
         0.000051507 "shairport.c:1741" interpolation soxr_delay_threshold is 30.
         0.000050042 "shairport.c:1742" resync time is 0.050000 seconds.
         0.000065361 "shairport.c:1743" allow a session to be interrupted: 0.
         0.000038574 "shairport.c:1744" busy timeout time is 120.
         0.000048280 "shairport.c:1745" drift tolerance is 0.001995 seconds.
         0.000049787 "shairport.c:1746" password is "(null)".
         0.000047658 "shairport.c:1747" ignore_volume_control is 0.
         0.000048671 "shairport.c:1751" volume_max_db is not set
         0.000038777 "shairport.c:1753" volume range in dB (zero means use the range specified by the mixer): 0.
         0.000064017 "shairport.c:1756" volume_range_combined_hardware_priority (1 means hardware mixer attenuation is used first) is 0.
         0.000068824 "shairport.c:1758" playback_mode is 0 (0-stereo, 1-mono, 1-reverse_stereo, 2-both_left, 3-both_right).
         0.000055037 "shairport.c:1759" disable_synchronization is 0.
         0.000051992 "shairport.c:1760" use_mmap_if_available is 1.
         0.000050778 "shairport.c:1762" output_format automatic selection is enabled.
         0.000037611 "shairport.c:1766" output_rate automatic selection is enabled.
         0.000047180 "shairport.c:1770" audio backend desired buffer length is 0.200000 seconds.
         0.000065964 "shairport.c:1772" audio_backend_buffer_interpolation_threshold_in_seconds is 0.120000 seconds.
         0.000051808 "shairport.c:1773" audio backend latency offset is 0.000000 seconds.
         0.000060366 "shairport.c:1775" audio backend silence lead-in time is -1.000000 seconds. A value -1.0 means use the default.
         0.000043369 "shairport.c:1776" zeroconf regtype is "_raop._tcp".
         0.000052695 "shairport.c:1777" decoders_supported field is 1.
         0.000063621 "shairport.c:1778" use_apple_decoder is 0.
         0.000050608 "shairport.c:1779" alsa_use_hardware_mute is 0.
         0.000048955 "shairport.c:1783" no special mdns service interface was requested.
         0.000080025 "shairport.c:1787" configuration file name "../var/shairport-sync.conf" resolves to "/volume1/@appstore/shairport-sync/var/shairport-sync.conf".
         0.000073261 "shairport.c:1816" loudness is 0.
         0.000055192 "shairport.c:1817" loudness reference level is -20.000000
         0.001010237 "mdns.c:119" Can't start a DACP monitor -- no mdns_dacp_monitor start registered.
         0.000102511 "audio_alsa.c:1940" keep_dac_busy is now "no"
         0.003029413 "tinysvcmdns.c:1532" sending announce for _raop._tcp.local.

         0.000684483 "tinysvcmdns.c:1504" data from=192.168.0.227 size=605

         0.000029749 "tinysvcmdns.c:756" rr_data_len goes beyond packet buffer: 29795 > 583

         0.000017874 "tinysvcmdns.c:894" error parsing answer #0

```

it looks like this is the problem:
```
 0.001010237 "mdns.c:119" Can't start a DACP monitor -- no mdns_dacp_monitor start registered.
```

ymartin59 commented on spk/shairport-sync/src/shairport-sync.sc:
```diff
@@ -8,4 +8,4 @@ dst.ports="8303/tcp"
 desc="Shairport Sync UDP"
 port_forward="yes"
-dst.ports="8303:8313/udp"
+dst.ports="8304:8313/udp"
```
No idea if this is temporary or definitive - but in any case, do not forget to update port documented at: https://github.com/SynoCommunity/spksrc/wiki/SynoCommunity-Used-Ports

hgy59 committed:
libsndfile: update to v1.0.28

hgy59 committed:
shairport-sync: update to v3.3.8

- update shairport-sync
- adjust config for DSM 7
- add soxr support
- add convolution filter support except for old compilers
- migrate from openssl to mbedtls

hgy59 committed:
define DSM 7 compatible sysconfdir

hgy59 commented:
Still no success:
```
ash-4.3# ./shairport-sync -vvv --configfile=../var/shairport-sync.conf
         0.000324300 "shairport.c:461" looking for configuration file at full path "/volume1/@appstore/shairport-sync/var/shairport-sync.conf"
         0.006382782 "shairport.c:1729" Started!
         0.000067335 "shairport.c:1756" software version: "3.3.8-libdaemon-mbedTLS-tinysvcmdns-external_mdns-ALSA-soxr-convolution-metadata-sysconfdir:/var/packages/shairport-sync/var"
         0.000023148 "shairport.c:1762" log verbosity is 3.
         0.000052853 "audio_alsa.c:998" alsa: init() -- alsa_backend_state => abm_disconnected.
         0.000018185 "audio_alsa.c:1015" alsa: alsa_maximum_stall_time of 0.200000 sec.
         0.000067371 "audio_alsa.c:1302" alsa: disable_standby_mode is "never".
         0.000016260 "audio_alsa.c:1304" alsa: disable_standby_mode_silence_threshold is 0.040000 seconds.
         0.000016852 "audio_alsa.c:1306" alsa: disable_standby_mode_silence_scan_interval is 0.004000 seconds.
         0.000038111 "audio_alsa.c:1345" alsa: output device name is "hw:USB".
         0.000135743 "shairport.c:1777" The processor is running little-endian.
         0.000011556 "shairport.c:1810" disable resend requests is off.
         0.000016519 "shairport.c:1814" diagnostic_drop_packet_fraction is 0.000000. A value of 0.0 means no packets will be dropped deliberately.
         0.000017926 "shairport.c:1815" statistics_requester status is 0.
         0.000015074 "shairport.c:1817" daemon status is 0.
         0.000016000 "shairport.c:1818" daemon pid file path is "/var/packages/shairport-sync/target/var/shairport-sync.pid".
         0.000013778 "shairport.c:1820" rtsp listening port is 8303.
         0.000013223 "shairport.c:1821" udp base port is 8304.
         0.000014629 "shairport.c:1822" udp port range is 10.
         0.000016815 "shairport.c:1823" player name is "Shairport Sync 3.3.8 on ds-arm".
         0.000015630 "shairport.c:1824" backend is "alsa".
         0.000014852 "shairport.c:1825" run_this_before_play_begins action is "(null)".
         0.000013741 "shairport.c:1826" run_this_after_play_ends action is "(null)".
         0.000016370 "audio_alsa.c:1968" keep_dac_busy is now "no"
         0.000066557 "shairport.c:1827" wait-cmd status is 0.
         0.000016703 "shairport.c:1828" run_this_before_play_begins may return output is 0.
         0.000014149 "shairport.c:1829" run_this_if_an_unfixable_error_is_detected action is "(null)".
         0.000014148 "shairport.c:1830" run_this_before_entering_active_state action is  "(null)".
         0.000013741 "shairport.c:1831" run_this_after_exiting_active_state action is  "(null)".
         0.000013408 "shairport.c:1832" active_state_timeout is  10.000000 seconds.
         0.000019704 "shairport.c:1833" mdns backend "(null)".
         0.000012703 "shairport.c:1834" userSuppliedLatency is 0.
         0.000015593 "shairport.c:1837" interpolation setting is "auto".
         0.000015111 "shairport.c:1838" interpolation soxr_delay_threshold is 30.
         0.000013630 "shairport.c:1839" resync time is 0.050000 seconds.
         0.000015260 "shairport.c:1840" allow a session to be interrupted: 0.
         0.000013333 "shairport.c:1841" busy timeout time is 120.
         0.000013593 "shairport.c:1842" drift tolerance is 0.001995 seconds.
         0.000013667 "shairport.c:1843" password is "(null)".
         0.000013074 "shairport.c:1844" ignore_volume_control is 0.
         0.000013445 "shairport.c:1848" volume_max_db is not set
         0.000013407 "shairport.c:1850" volume range in dB (zero means use the range specified by the mixer): 0.
         0.000013667 "shairport.c:1854" volume_range_combined_hardware_priority (1 means hardware mixer attenuation is used first) is 0.
         0.000013667 "shairport.c:1856" playback_mode is 0 (0-stereo, 1-mono, 1-reverse_stereo, 2-both_left, 3-both_right).
         0.000016222 "shairport.c:1857" disable_synchronization is 0.
         0.000015519 "shairport.c:1858" use_mmap_if_available is 1.
         0.000012852 "shairport.c:1860" output_format automatic selection is enabled.
         0.000013334 "shairport.c:1864" output_rate automatic selection is enabled.
         0.000012703 "shairport.c:1868" audio backend desired buffer length is 0.200000 seconds.
         0.000016927 "shairport.c:1870" audio_backend_buffer_interpolation_threshold_in_seconds is 0.120000 seconds.
         0.000017370 "shairport.c:1871" audio backend latency offset is 0.000000 seconds.
         0.000016815 "shairport.c:1873" audio backend silence lead-in time is "auto".
         0.000013000 "shairport.c:1877" zeroconf regtype is "_raop._tcp".
         0.000015778 "shairport.c:1878" decoders_supported field is 1.
         0.000014889 "shairport.c:1879" use_apple_decoder is 0.
         0.000012815 "shairport.c:1880" alsa_use_hardware_mute is 0.
         0.000012926 "shairport.c:1884" no special mdns service interface was requested.
         0.000049334 "shairport.c:1888" configuration file name "../var/shairport-sync.conf" resolves to "/volume1/@appstore/shairport-sync/var/shairport-sync.conf".
         0.000018964 "shairport.c:1894" metadata enabled is 1.
         0.000018259 "shairport.c:1895" metadata pipename is "/tmp/shairport-sync-metadata".
         0.000015593 "shairport.c:1897" metadata socket address is "(null)" port 0.
         0.000013889 "shairport.c:1898" metadata socket packet size is "500".
         0.000013074 "shairport.c:1899" get-coverart is 1.
         0.000012667 "shairport.c:1912" convolution is 0.
         0.000013000 "shairport.c:1913" convolution IR file is "(null)"
         0.000013556 "shairport.c:1914" convolution max length 8192
         0.000013111 "shairport.c:1915" convolution gain is 0.000000
         0.000013111 "shairport.c:1917" loudness is 0.
         0.000013445 "shairport.c:1918" loudness reference level is -20.000000
         0.000150409 "rtsp.c:1665" metadata pipe name is "/tmp/shairport-sync-metadata".
         0.000112409 "rtsp.c:158" Creating metadata queue "pipe".
         0.000123816 "rtsp.c:158" Creating metadata queue "multicast".
         0.000878752 "mdns.c:119" Can't start a DACP monitor -- no mdns_dacp_monitor start registered.
         0.000013259 "tinysvcmdns.c:1532" sending announce for _raop._tcp.local.
         0.002993259 "tinysvcmdns.c:1504" data from=192.168.0.229 size=390
         0.000079149 "tinysvcmdns.c:756" rr_data_len goes beyond packet buffer: 29795 > 368
         0.000022408 "tinysvcmdns.c:894" error parsing answer #0
         1.501652247 "shairport.c:194" soxr_delay_index: 8.
         0.000082038 "shairport.c:201" "soxr" interpolation has been chosen.
```

Running as service puts the following to `/var/log/messages`:
```
2021-12-04T23:13:12+01:00 ds-arm shairport-sync[29608]: warning: alsa: error -19 ("No such device") opening alsa device "hw:USB".
2021-12-04T23:13:12+01:00 ds-arm shairport-sync[29608]: warning: alsa: error -19 ("No such device") opening alsa device "hw:USB".
2021-12-04T23:13:14+01:00 ds-arm shairport-sync[29608]: warning: alsa: error -19 ("No such device") opening alsa device "hw:USB".
2021-12-04T23:13:43+01:00 ds-arm shairport-sync: SYSTEM:        Last message 'warning: alsa: error' repeated 3626 times, suppressed by syslog-ng on ds-arm
2021-12-04T23:13:43+01:00 ds-arm shairport-sync[29608]: warning: alsa: error -19 ("No such device") opening alsa device "hw:USB".
2021-12-04T23:13:43+01:00 ds-arm shairport-sync[29608]: warning: alsa: error -19 ("No such device") opening alsa device "hw:USB".
2021-12-04T23:13:43+01:00 ds-arm shairport-sync[29608]: warning: alsa: error -19 ("No such device") opening alsa device "hw:USB".
2021-12-04T23:14:14+01:00 ds-arm shairport-sync: SYSTEM:        Last message 'warning: alsa: error' repeated 3882 times, suppressed by syslog-ng on ds-arm
2021-12-04T23:14:14+01:00 ds-arm shairport-sync[29608]: warning: alsa: error -19 ("No such device") opening alsa device "hw:USB".
2021-12-04T23:14:14+01:00 ds-arm shairport-sync[29608]: warning: alsa: error -19 ("No such device") opening alsa device "hw:USB".
2021-12-04T23:14:14+01:00 ds-arm shairport-sync[29608]: warning: alsa: error -19 ("No such device") opening alsa device "hw:USB".
```

So I suppose the service must be run as root, and this will forbidden on DSM 7.

hgy59 committed:
adjust libsndfile/Makefile

hgy59 committed:
update shairport-sync to v3.3.9

blendR8 commented:
the spk from https://github.com/SynoCommunity/spksrc/pull/3866#issuecomment-653742783 wors for me after changing the device to hw:Device and giving /dev/snd recursively chmod 777
so it is a permission problem.
Iam on DSM 6.1

blendR8 commented:
can you build an spk with version 3.3.9 ? @hgy59

hgy59 commented:
> can you build an spk with version 3.3.9 ? @hgy59

I can rebase this branch and push so the github action will create the packages. But it does not support DSM 6.1 anymore. Since Synology retired the older toolchains we build for DSM 6 for Version 6.2.4 only.

But I still have those toolchains locally. What is you model? Can you update to DSM 6.2.4?

blendR8 commented:
I am on 6.1.7 but currently I can’t update to 6.2. do you mean that you can’t build a version prior 6.2.4?

hgy59 commented:
> I am on 6.1.7 but currently I can’t update to 6.2. do you mean that you can’t build a version prior 6.2.4?

What is your model? DSM 6.2.4 is available for almost all models (except EDS14, limitid to DSM 6.0.3-8754).

I can locally build for DSM 6.1 with the 6.1 toolchains in my cache.

hgy59 commented:
> I am on 6.1.7 but currently I can’t update to 6.2. do you mean that you can’t build a version prior 6.2.4?

you find it [here](https://github.com/hgy59/spksrc/releases/tag/shairport-sync_3.3.9)

blendR8 commented:
I will try and give feedback. Thank you so much

blendR8 commented:
It worked now for more then a week. no crashed or other problems. again: thank you so much

hgy59 commented:
> It worked now for more then a week. no crashed or other problems. again: thank you so much

@blendR8 thank you for the feedback.

Just for clarification: do you stream to speakers attached to the Diskstation?

hgy59 committed:
shairport-sync: update to v4.3.2

- update mbedtls to v2.28.6
- adjust config (daemon_pid_dir is still required, despite it got removed in sample config)

hgy59 committed:
shairport-sync: update to v4.3.7

mreid-tt commented:
hey @cortices, @blendR8 — I’ve made another pass at modernizing the shairport-sync package in #6950. If you have some time, you can download and install it from the build artifacts and share any feedback. Instructions for installing pre-release packages are available in our [wiki](https://github.com/SynoCommunity/spksrc/wiki/Frequently-Asked-Questions#download-development-and-pre-release-packages-created-by-github-build-action).
